apps/dashboard/app/controllers/files_controller.rb

```diff
@@ -4,6 +4,9 @@ def fs
     request.format = 'json' if request.headers['HTTP_ACCEPT'].split(',').include?('application/json')
 
     @path = normalized_path
+    AllowlistPolicy.default.validate!(@path)
```

Comment:
We can TODO this, but I wonder if we should refactor this out to a `preaction` instead of having exceptions as a control flow (except for `edit`)

Reply:
Yes a before_action was what I wanted to do first but before I do it, the JSON requests should all return error codes in error states (right now only the upload does) but the fetch logic gets more complex then (though that only needs to be completed once). Also I guess the only page we display right now is the index page - that would also be our error page if HTML is requested, with the alert set and everything else empty.

It was enough work to postpone.